feat(transactions): added pagination to transactions

NuttyShrimp opened the pull request:
First time writing in this language so I'm open for criticism.

Adds the `limit=` and `start=` query parameters to the `/api/v1/users/USERNAME/transactions` endpoint.

closes #114

TomNaessens commented:
Would be nice to have these tested and added to the API specs at https://github.com/ZeusWPI/Tab/blob/master/spec/integration/api/v1/transactions_controller_spec.rb. This spec will automatically generate a swagger so users are aware the params exist.

It's not explicitly needed to check for the params existence, using `limit` and `offset` with `nil` will just return everything. 
Looking good otherwise!

NuttyShrimp commented:
i've added the test (after some struggles) and also added ordering to the transactions. It think that's logical change to prevent users from getting unordered list's but I'm not sure if it is required to explicitly order by id
